### PR TITLE
DF-815-<fieldset> structures within the catalogue search results page.

### PR DESCRIPTION
**Pull request by JohnHeeryTNA**
Ticket URL: https://national-archives.atlassian.net/jira/software/projects/DF/issues/DF-815

## About these changes

Implementing <fieldset> tags around 'To' and 'From' date ranges on catalogue search results page 

## How to check these changes

Perform a search, go to catalogue results, filter by date ranges to ensure this is working correctly.

## Before assigning to reviewer, please make sure you have

- [x] Checked things thoroughly before handing over to reviewer.
- [x] Checked PR title starts with ticket number as per project conventions to help us keep track of changes.
- [x] Ensured that PR includes only commits relevant to the ticket.
- [x] Waited for all CI jobs to pass before requesting a review.
- [x] Added/updated tests and documentation where relevant.

## Merging PR guidance

Follow [docs\developer-guide\contributing.md](https://nationalarchives.github.io/ds-wagtail/developer-guide/contributing/)

## Deployment guidance

Follow [docs\infra\environments.md](https://nationalarchives.github.io/ds-wagtail/infra/environments/)

### CODE REVIEW

File: templates/search/blocks/search_filters.html [hunk 1]
```diff
@@ -111,6 +114,7 @@ <h4 class="tna-heading search-filters__accordion-section-heading" id="record_ope
                         </legend>
 
                         <div class="search-filters__opening-date-section">
+                            <fieldset>
                             <span class="example-text">For example, 27 3 2007</span><br>
```
Comment (ahosgood) [inline templates/search/blocks/search_filters.html:118]: I think each of these need a unique `id` so we can add `aria-describedby="whatever-id"` to the `<fieldset>`

File: templates/search/blocks/search_filters.html [hunk 2]
```diff
@@ -64,9 +64,11 @@ <h4 class="tna-heading search-filters__accordion-section-heading" id="record_cov
                                 </div>
                             {% endif %}
                             {{ form.covering_date_from }}
+                            </fieldset>
                         </div>
 
                         <div class="search-filters__creation-date-section">
+                            <fieldset>
```
[the remark this answers is not in the document]
Comment (JohnHeeryTNA) [reply templates/search/blocks/search_filters.html]: I've added in `role="group`" to these `<fieldset>`

File: templates/search/blocks/search_filters.html (hunk 1 above)
Comment (JohnHeeryTNA) [reply templates/search/blocks/search_filters.html:118]: id's have now been applied

File: templates/search/blocks/search_filters.html [hunk 3]
```diff
@@ -54,7 +54,7 @@ <h4 class="tna-heading search-filters__accordion-section-heading" id="record_cov
                         <div>{{ form.covering_date_from.label_tag }}</div>
 
                         <div class="search-filters__creation-date-section">
-                            
+                            <fieldset>
```
[the remark this answers is not in the document]
Comment (JohnHeeryTNA) [reply templates/search/blocks/search_filters.html]: That `class` isn't actually referenced within the css, so I have removed it.

File: templates/search/blocks/search_filters.html [hunk 4]
```diff
@@ -45,16 +45,16 @@ <h3 class="tna-heading sr-only">Edit filters</h3>
...
                             <h4 class="tna-heading search-filters__accordion-section-heading" id="record_covering_date">Dates</h4>
                         </legend>
 
                         <span class="example-text">For example, 27 3 2007</span>
                         <div>{{ form.covering_date_from.label_tag }}</div>
 
-                        <div class="search-filters__creation-date-section">
-                            
+                        <div id="from-date">
+                            <fieldset role="group" aria-describedby="from-date">
```
Comment (ahosgood) [inline templates/search/blocks/search_filters.html:57]: Sorry - I don't think I described this very well!

The `aria-describedby="from-date"` needs to be added to the `<fieldset>` on line 48 and then remove these nested `<fieldset>`s.

I'm taking this advice from https://design-system.service.gov.uk/components/date-input/

Comment (JohnHeeryTNA) [reply templates/search/blocks/search_filters.html]: I've now removed the nested `<fieldset>`s and adjusted the `aria-describedby` elements

File: templates/search/blocks/search_filters.html [hunk 5]
```diff
@@ -44,17 +44,17 @@ <h2 class="tna-heading search-filters__heading">Refine results</h2>
...
             {% if form.group.value == 'tna' or form.group.value == 'digitised' or form.group.value == 'nonTna' %}
-                <div class="search-filters__accordion-section">
-                    <fieldset>
+                <div class="search-filters__accordion-section" id="from-and-to-dates">
+                    <fieldset role="group" aria-describedby="from-and-to-dates">
                         <legend>
                             <h4 class="tna-heading search-filters__accordion-section-heading" id="record_covering_date">Dates</h4>
                         </legend>
 
                         <span class="example-text">For example, 27 3 2007</span>
```
Comment (ahosgood) [inline templates/search/blocks/search_filters.html:53]:
```suggestion
                <div class="search-filters__accordion-section">
                    <fieldset role="group" aria-describedby="from-date">
                        <legend>
                            <h4 class="tna-heading search-filters__accordion-section-heading" id="record_covering_date">Dates</h4>
                        </legend>

                        <span class="example-text" id="from-date">For example, 27 3 2007</span>
```

Comment (ahosgood) [reply templates/search/blocks/search_filters.html:53]: This is what I meant

File: templates/search/blocks/search_filters.html [hunk 6]
```diff
@@ -104,13 +107,14 @@ <h4 class="tna-heading search-filters__accordion-section-heading" id="record_cov
...
+                <div class="search-filters__accordion-section" id="opening-and-closing-dates">
+                    <fieldset role="group" aria-describedby="opening-and-closing-dates">
                         <legend>
                             <h4 class="tna-heading search-filters__accordion-section-heading" id="record_opening_date">Record opening date</h4>
                         </legend>
 
-                        <div class="search-filters__opening-date-section">
+                        <div class="search-filters__opening-date-section" id="opening-start-date">
+
                             <span class="example-text">For example, 27 3 2007</span><br>
```
Comment (ahosgood) [inline templates/search/blocks/search_filters.html:118]:
```suggestion
                            <span class="example-text" id="opening-start-date">For example, 27 3 2007</span><br>
```

File: templates/search/blocks/search_filters.html [hunk 7]
```diff
@@ -104,13 +107,14 @@ <h4 class="tna-heading search-filters__accordion-section-heading" id="record_cov
...
-                <div class="search-filters__accordion-section">
-                    <fieldset>
+                <div class="search-filters__accordion-section" id="opening-and-closing-dates">
+                    <fieldset role="group" aria-describedby="opening-and-closing-dates">
                         <legend>
                             <h4 class="tna-heading search-filters__accordion-section-heading" id="record_opening_date">Record opening date</h4>
                         </legend>
 
-                        <div class="search-filters__opening-date-section">
+                        <div class="search-filters__opening-date-section" id="opening-start-date">
```
Comment (ahosgood) [inline templates/search/blocks/search_filters.html:116]:
```suggestion
                        <div class="search-filters__opening-date-section">
```

File: templates/search/blocks/search_filters.html [hunk 8]
```diff
@@ -104,13 +107,14 @@ <h4 class="tna-heading search-filters__accordion-section-heading" id="record_cov
 
 
             {% if form.group.value == 'tna' or form.group.value == 'digitised' %}
-                <div class="search-filters__accordion-section">
-                    <fieldset>
+                <div class="search-filters__accordion-section" id="opening-and-closing-dates">
+                    <fieldset role="group" aria-describedby="opening-and-closing-dates">
```
Comment (ahosgood) [inline templates/search/blocks/search_filters.html:111]:
```suggestion
                    <fieldset role="group" aria-describedby="opening-start-date">
```

**Review by ahosgood — APPROVED** [review]

**Pull request merged**